File: backend/app/schemas/ticket.py
# ============================================================
# 工单相关 Pydantic Schema
# ============================================================

from datetime import datetime
from typing import Optional

from pydantic import BaseModel, Field


class TicketCreateRequest(BaseModel):
    """创建工单请求（员工）"""
    title: str = Field(..., min_length=1, max_length=256, description="工单标题")
    question: str = Field(..., min_length=1, description="原始问题")
    detail: Optional[str] = Field(default=None, max_length=2000, description="补充说明")
    session_id: Optional[str] = Field(default=None, max_length=64, description="关联会话ID")
    priority: str = Field(default="medium", description="优先级：high/medium/low")


class TicketReplyRequest(BaseModel):
    """处理工单请求（管理员）"""
    reply: str = Field(..., min_length=1, max_length=5000, description="处理回复")


class TicketStatusRequest(BaseModel):
    """更新工单状态请求（管理员）"""
    status: str = Field(..., description="状态：pending/processing/resolved/closed")


class TicketInfo(BaseModel):
    """工单信息响应"""
    id: int
    title: str
    question: str
    detail: Optional[str] = None
    status: str
    priority: str
    reply: Optional[str] = None
    reply_count: int
    session_id: Optional[str] = None
    created_by: int
    handler_id: Optional[int] = None
    created_at: datetime
    updated_at: Optional[datetime] = None
    resolved_at: Optional[datetime] = None

    # 关联显示字段（由 JOIN 填充）
    creator_name: Optional[str] = None
    handler_name: Optional[str] = None

    class Config:
        from_attributes = True

File: frontend/src/api/ticket.ts
import client from './client'

export interface Ticket {
  id: number
  title: string
  question: string
  detail: string | null
  status: string
  priority: string
  reply: string | null
  reply_count: number
  session_id: string | null
  created_by: number
  handler_id: number | null
  created_at: string
  updated_at: string | null
  resolved_at: string | null
  creator_name: string | null
  handler_name: string | null
}

export const ticketApi = {
  // 员工端
  create: (data: { title: string; question: string; detail?: string; session_id?: string; priority?: string }) =>
    client.post('/tickets', data),
  myTickets: (params: { page?: number; page_size?: number; status?: string } = {}) =>
    client.get('/tickets', { params }),
  getTicket: (id: number) => client.get(`/tickets/${id}`),

  // 管理端
  listAll: (params: { page?: number; page_size?: number; status?: string; priority?: string; keyword?: string } = {}) =>
    client.get('/admin/tickets', { params }),
  getStats: () => client.get('/admin/tickets/stats'),
  handle: (id: number, reply: string) => client.put(`/admin/tickets/${id}/handle`, { reply }),
  updateStatus: (id: number, status: string) => client.put(`/admin/tickets/${id}/status`, { status }),
}

export const TICKET_STATUS_LABELS: Record<string, { label: string; type: 'info' | 'warning' | 'success' | 'danger' }> = {
  pending: { label: '待处理', type: 'warning' },
  processing: { label: '处理中', type: 'info' },
  resolved: { label: '已解决', type: 'success' },
  closed: { label: '已关闭', type: 'danger' },
}

export const TICKET_PRIORITY_LABELS: Record<string, { label: string; type: 'danger' | 'warning' | 'success' }> = {
  high: { label: '高', type: 'danger' },
  medium: { label: '中', type: 'warning' },
  low: { label: '低', type: 'success' },
}